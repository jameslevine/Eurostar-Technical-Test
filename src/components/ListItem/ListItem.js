import React from "react";
import styled from "styled-components";

const ListItem = props => {
  return (
    <ListItemContainer>
      <ListItemElement>
        <Title>
          {props.title} - {props.identification}
        </Title>
      </ListItemElement>
    </ListItemContainer>
  );
};

export default ListItem;

const Title = styled.h3`
  font-weight: 600;
  margin: 0;
`;

const ListItemElement = styled.li`
  list-style: none;
`;

const ListItemContainer = styled.div`
  border: 1px solid black;
  padding: 1px;
  margin: 1em 0;
`;
