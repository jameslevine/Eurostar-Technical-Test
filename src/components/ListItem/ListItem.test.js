import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import ListItem from "./ListItem";
const mockStore = configureStore([]);

describe("My Connected React-Redux Component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      list: {
        total: 200
      }
    });
    component = renderer.create(
      <Provider store={store}>
        <ListItem />
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
