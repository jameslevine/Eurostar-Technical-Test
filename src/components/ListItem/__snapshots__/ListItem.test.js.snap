// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="sc-htpNat fMPSta"
>
  <li
    className="sc-bwzfXH fHycKW"
  >
    <div>
      <h3
        className="sc-bdVaJa fvCAJY"
      >
        Title: 
      </h3>
      <h3
        className="sc-bdVaJa fvCAJY"
      >
         of 
        200
      </h3>
    </div>
    <div
      className="sc-bxivhb ezPioi"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    />
  </li>
</div>
`;
